### ISSUE TITLE
Wrong SQL generated for nested structs

### ISSUE THREAD

**Issue by ricochet1k** [go-pg/pg#383]
```go
type Model struct {
	ID int
}

type MajorVersion struct {
	TableName struct{} `sql:"api_majorversion"`
	Model
	Value int64
	Notes string
}

type MinorVersion struct {
	TableName struct{} `sql:"api_minorversion"`
	Model
	Value int64
	Major MajorVersion
	MajorID int
	Notes   string
}

type MaintenanceVersion struct {
	TableName struct{} `sql:"api_maintenanceversion"`
	Model
	Value   int64
	Minor   MinorVersion
	MinorID int
	Notes   string
}

var maint api.MaintenanceVersion
err = db.Model(&maint).Column("maintenance_version.*", "Minor", "Minor.Major").Limit(1).Select()
```
```
2016/10/27 15:12:04 select_test.go:55: SELECT "maintenance_version".*, "minor"."id" AS "minor__id", "minor"."value" AS "minor__value", "minor"."major_id" AS "minor__major_id", "minor"."notes" AS "minor__notes", "major"."id" AS "minor__major__id", "major"."value" AS "minor__major__value", "major"."notes" AS "minor__major__notes" FROM api_maintenanceversion AS "maintenance_version" LEFT JOIN api_minorversion AS "minor" ON "minor"."id" = "maintenance_version"."minor_id" LEFT JOIN api_majorversion AS "major" ON "major"."id" = "minor_version"."major_id" LIMIT 1
--- FAIL: TestSelect (0.16s)
select_test.go:57: Unexpected error: ERROR #42P01 missing FROM-clause entry for table "minor_version" 
```

It looks like "minor_version"."major_id" is using the struct table name instead of the field table name.

**Comment by vmihailenco** [comment]
I have a fix in progress, but this revealed a bigger issue so I need more time to think about it.

**Comment by vmihailenco** [comment]
Hopefully fixed in v5.0.4. That was a big change so I am not sure I got all the details right.
